cli/src/unimarkup.rs

```diff
@@ -22,17 +19,15 @@ use crate::log_id::{GeneralErrLogId, GeneralInfLogId, CLI_LOG_ID_MAP};
 ///
 /// # Errors
 ///
-/// Returns a [`MappedLogId`] if error occurs during compilation.
-pub fn compile(config: Config) -> Result<(), MappedLogId> {
-    let source = fs::read_to_string(&config.input).map_err(|err| {
-        (GeneralErrLogId::FailedReadingFile as LogId)
-            .set_event_with(
-                &CLI_LOG_ID_MAP,
-                &format!("Could not read file: '{:?}'", &config.input),
-                file!(),
-                line!(),
-            )
-            .add_info(&format!("Cause: {}", err))
+/// Returns a [`GeneralError`] if error occurs during compilation.
+pub fn compile(config: Config) -> Result<(), GeneralError> {
+    let source = fs::read_to_string(&config.input).map_err(|error| {
+        log!(
+            GeneralError::FailedReadingFile,
+            &format!("Could not read file: '{:?}'", &config.input),
+            add: AddonKind::Info(format!("Cause: {}", error))
+        );
+        GeneralError::FailedReadingFile
```

Comment:
Thanks for the tip with `dbg!()`.
I decided to keep `log!()` as is, since it already returns a `LogEvent`.
To solve your request, I added the `pipe!()` macro that behaves like `log!()`, but "pipes" the `LogId` like it is done in `dbg!()`.